crates/cards/src/validate.rs

```diff
@@ -30,12 +32,25 @@ impl FromStr for CardNumber {
 }
 
 #[allow(dead_code)]
-impl CardNumber {
-    fn get_secret(&self) -> &StrongSecret<String, CardNumberStrategy> {
+impl Deref for CardNumber {
+    type Target = StrongSecret<String, CardNumberStrategy>;
+
+    fn deref(&self) -> &StrongSecret<String, CardNumberStrategy> {
         &self.0
     }
 }
 
+fn from_str<'de, D>(deserializer: D) -> Result<StrongSecret<String, CardNumberStrategy>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: String = Deserialize::deserialize(deserializer)?;
+    match CardNumber::from_str(&s) {
+        Ok(secret) => Ok(secret.clone()),
+        Err(e) => Err(serde::de::Error::custom(e.to_string())),
+    }
+}
```

Comment:
Can you manually implement `Deserialize` for `CardNumber` instead? Something along the lines of:

```rust
str::deserialize(deserializer)?.try_into().map_err(de::Error::custom)
```

Reply:
I currently did it via:

```rust
deserialize<D: Deserializer<'de>>(d: D) -> Result<Self, D::Error> {
        let s = String::deserialize(d)?;
        match Self::from_str(&s) {
            Ok(card_number) => Ok(card_number),
            Err(e) => Err(serde::de::Error::custom(e.to_string())),
        }
    }
```

If I want to do it via `try_into()` I would also have to implement `TryFrom<&str>` or `TryFrom<String>` for `CardNumber` right? And in there also use the card number validation.

Is the `try_into()` variant preferred?

Reply:
It's fine, the `try_into()` is not required.